### PR TITLE
Enable Autocheckpoint and exit after saving an autocheckpoint

### PR DESCRIPTION
**Pull request by xuefgu**
This PR does two things:

1) Enable Autocheckpoint in MaxText. Along the way, the checkpoint manager is initialized in a more canonical way, with the option `save_interval_steps` passed as an option during creation.

2) Make the program exit after the checkpoint has been saved.

I have tested this PR to make sure that the full functionality of Autocheckpoint works as desired on two v4-8: a checkpointing process is automatically started upon a maintenance event, the program exits after saving the checkpoint, and the TPU goes through maintenance thereafter.

### CODE REVIEW

**Review by gobbleturk — APPROVED** [review]
This looks great to me, thank you for adding this!

**Review by rwitten — APPROVED** [review]
This looks great to me and see Matt has already approved.

I assume you already have it but has Colin Gaffney confirmed (I don't need an approval in the system).

Has this been tested? (Or are we comfortable with our XLML tests -- I'll let you and Matt decide.)

**Comment by xuefgu** [comment]
> This looks great to me and see Matt has already approved.
> 
> I assume you already have it but has Colin Gaffney confirmed (I don't need an approval in the system).
> 
> Has this been tested? (Or are we comfortable with our XLML tests -- I'll let you and Matt decide.)

**Comment by xuefgu** [comment]
Colin has reviewed this PR.

Matt and I confirmed that this should not change XLML tests behavior.

**Pull request merged**
